### PR TITLE
chip-tool: do not hardcode Thread network id

### PR DESCRIPTION
**Pull request by bluebin14**
#### Problem
What is being fixed?  Examples:
* chip-tool hardcodes Thread network id to 0123456789abcdef. This does not work in most cases, see issue #8097

#### Change overview
What's in this PR
* use the Thread network id of the dataset instead of the hardcoded network id
* not having the extendedPanId in the dataset is now an error, as it should be, since it is a required item for looking up the dataset (note however that it won't be missing though since code will fail already at the add Thread network command if it is missing).

#### Testing
How was this tested? (at least one bullet point required)
* tested manually using chip-tool and HomePod Mini where Thread network id cannot be hardcoded

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

File: examples/chip-tool/commands/pairing/PairingCommand.cpp [hunk 1]
```diff
@@ -222,6 +222,25 @@ CHIP_ERROR PairingCommand::AddWiFiNetwork()
...
 }
 
+chip::ByteSpan PairingCommand::GetThreadNetworkId()
+{
+    // For Thread devices the networkId is the extendedPanId and it is
+    // part of the dataset defined by OpenThread
+
+    Thread::OperationalDataset dataset;
+    CHIP_ERROR err = dataset.Init(mOperationalDataset);
+    if (err == CHIP_NO_ERROR)
```
Comment (andy31415) [inline examples/chip-tool/commands/pairing/PairingCommand.cpp:232]: nit: I believe if we invert the conditions, we could reduce nesting:

```cpp
Thread::OperationalDataset dataset;

if (dataset.Init(mOperationalDataSet) != CHIP_NO_ERROR) {
  return ByteSpan();
}

if (dataset.GetExtendedPanId(mExtendedPanId) != CHIP_NO_ERROR) {
  return ByteSpan();
}

return ByteSpan(mExtendedPanId);
```

File: examples/chip-tool/commands/pairing/PairingCommand.h [hunk 2]
```diff
@@ -138,6 +140,7 @@ class PairingCommand : public Command,
     uint16_t mDiscriminator;
     uint32_t mSetupPINCode;
     chip::ByteSpan mOperationalDataset;
+    uint8_t mExtendedPanId[8];
```
Comment (andy31415) [inline examples/chip-tool/commands/pairing/PairingCommand.h:143]: use `kSizeExtendedPanId` for sizing this.

**Comment by andy31415** [comment]
@bluebin14 for updates, please just add new commits instead of force pushing. This way it is easier to see deltas and the commits will get squashed-merged when the PR is approved anyway.

The only exception to this is changes involving ZAP, where we want the 'generate files' as a separate commit and in which case force push is better.

File: examples/chip-tool/commands/pairing/PairingCommand.cpp (hunk 1 above)
Comment (bluebin14) [reply examples/chip-tool/commands/pairing/PairingCommand.cpp]: Applied change.

File: examples/chip-tool/commands/pairing/PairingCommand.h (hunk 2 above)
Comment (bluebin14) [reply examples/chip-tool/commands/pairing/PairingCommand.h]: Done.

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**